### PR TITLE
Pull request for merging 

### PR DESCRIPTION
**Pull request by NitinChandak**

### CODE REVIEW

File: experiment/posttest.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[{"question":"1.This is a sample question:","answers":{"a":"This is a sample answer A","b":"This is a sample answer B"},"correctAnswer":"a"},{"question":"<img src='images/8.PNG'><br>Identify the location of Secondary electron detector","answers":{"a":"<img src='images/1b.png'>","b":"<img src='images/1a.png'>","c":"<img src='images/1c.PNG'>"},"correctAnswer":"c"}]
```
Comment (raj-vlabs) [inline experiment/posttest.json]: Remove sample questions

File: experiment/pretest.json [hunk 2]
```diff
@@ -0,0 +1 @@
+[{"question":"1.This is a sample question:","answers":{"a":"This is a sample answer A","b":"This is a sample answer B"},"correctAnswer":"a"},{"question":"<img src='images/8.PNG'><br>Identify the location of Secondary electron detector","answers":{"a":"<img src='images/1b.png'>","b":"<img src='images/1a.png'>","c":"<img src='images/1c.PNG'>"},"correctAnswer":"c"}]
```
Comment (raj-vlabs) [inline experiment/pretest.json]: Remove sample questions

File: experiment/references.md [hunk 3]
```diff
@@ -1 +1,17 @@
-### Link your references in here
\ No newline at end of file
+1. http://en.wikipedia.org/wiki/Surfactant
+
+2. http://nsdl.niscair.res.in/bitstream/123456789/311/1/PDF+corrected+surface+chemistry.pdf
+
+3. http://www.elmhurst.edu/~chm/vchembook/554soap.html
```
Comment (raj-vlabs) [inline experiment/references.md]: Please use https references only and ensure that all of them work. Also provide meaningful link descriptions in place of putting the complete link there

File: experiment/simulation/cleansingAction.html [hunk 4]
```diff
@@ -0,0 +1,62 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset = "utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+	<!-- Add CSS at the head of HTML file -->
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/animejs/3.2.1/anime.min.js" integrity="sha512-z4OUqw38qNLpn1libAN9BsoDx6nbNFio5lA6CuTp9NlK83b89hgyCVq+N5FdBJptINztxn1Z3SaKSKUS5UP60Q==" crossorigin="anonymous"></script>
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
```
Comment (raj-vlabs) [inline experiment/simulation/cleansingAction.html]: Bootstrap below version 4 is not approved. Please upgrade the version.

File: experiment/simulation/cleansingAction.html [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.1/js/bootstrap.min.js"></script>
+    <link rel="stylesheet" href="./css/main.css">
+    <title>Cleansing action</title>
+
+</head>
+    <body>
+		<!-- Your code goes here-->
```
Comment (raj-vlabs) [inline experiment/simulation/cleansingAction.html]: Remove this guide comment

File: experiment/simulation/css/main.css [hunk 6]
```diff
@@ -1 +1,226 @@
-/* You CSS goes in here */
\ No newline at end of file
+#main{
```
Comment (raj-vlabs) [inline experiment/simulation/css/main.css]: Please use classes instead of hard coding ids where possible.

File: experiment/simulation/images/tail.svg [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+	</defs>
+	<style>
+		tspan { white-space:pre }
+	</style>
+	<use id="Layer 2" href="#img1" x="399" y="7" />
+	<use id="Background" href="#img2" x="0" y="0" />
```
Comment (raj-vlabs) [inline experiment/simulation/images/tail.svg]: Please use lowercase identifiers

File: experiment/simulation/js/main.js [hunk 8]
```diff
@@ -1 +1,251 @@
-//Your JavaScript goes in here
+let detergentAddCounter = 1;
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js:2]: Please use strict mode.
I see that you are using it at the function level. Please use it at the top level too unless it breaks some existing library.

File: experiment/simulation/js/main.js [hunk 9]
```diff
@@ -1 +1,251 @@
...
+if(document.getElementById("detergent3")){
+    document.getElementById("detergent3").style.pointerEvents = 'none';
+}
+
+
+// Functions for structure of soap part.
+// below function shows information about micelle parts when hovered upon
+function showMicelleInformation(part){
+    let tailInfo = document.getElementById("micelle-information");
+    Info = "";
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js]: Where is declaration for Info and why is it capitalised?

**Review by raj-vlabs — CHANGES_REQUESTED** [review]
1. Please remove the sample questions from pretest and posttest. In case you are not able to get questions from the SME, please inform Priya and leave an empty array in the json files so your build goes through.
2. Please use https references only and ensure that all of them work. Also provide meaningful link descriptions.
3. Please use Bootstrap version 4 or above only.
4. Please remove all guide comments.
5. Prefer classes over ids for css styling
6. Please make sure you run the code through the style guidelines pinned on the channel before submitting it for review.

File: experiment/simulation/cleansingAction.html [hunk 10]
```diff
@@ -0,0 +1,63 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset = "utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/animejs/3.2.1/anime.min.js" integrity="sha512-z4OUqw38qNLpn1libAN9BsoDx6nbNFio5lA6CuTp9NlK83b89hgyCVq+N5FdBJptINztxn1Z3SaKSKUS5UP60Q==" crossorigin="anonymous"></script>
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css" integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm" crossorigin="anonymous">
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
+    <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
```
Comment (raj-vlabs) [inline experiment/simulation/cleansingAction.html]: Why do we need both jQuery and jQuery slim?

File: experiment/simulation/cleansingAction.html [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+            <img src="./images/bench.png" class="bench"/>
+            <img src="./images/spoon.png" class="spoon1" onclick="moveSpoon()"/>
+            <img src="./images/Detergent.png" class="detergent1" onclick="moveSpoon()"/>
+            <img src="./images/glassRod.png" class="glassRod" onclick="moveGlassRod('cleansingAction')"/>
+            <img src="./images/clothWithStain.png" class="clothWithStain" onclick="moveClothWithStain()"/> 
+            <img src="./images/plainCloth.png" class="plainCloth"/>
+            <div class="molecularView">
+                <div class="videoMessage">Molecular view of soap molecules in water:</div>  
+                <video class="videos"  width="100%" height="240"  autoplay	muted loop>
+                    <source class="videoSource" src="./images/videos/MolecularView.mp4" type="video/mp4"/>
```
Comment (raj-vlabs) [inline experiment/simulation/cleansingAction.html:46]: What is the size of this video file? Is it necessary to autoplay this? Can we think about lazy loading the video only when the user requests for it?

File: experiment/simulation/css/main.css [hunk 12]
```diff
@@ -1 +1,228 @@
...
+}
+
+.title {
+    text-align: center;
+    margin-top: 5px;
+    font-size: 2.5vw;
+    z-index: -1;
+}
+
+.nav-link ,.navbar-nav,.navbar ,.navbar-custom{
```
Comment (raj-vlabs) [inline experiment/simulation/css/main.css]: It'll be better to move the background part into its separate class  like nav-bg. For padding you can use bootstrap built in padding classes https://getbootstrap.com/docs/4.0/utilities/spacing/

File: experiment/simulation/css/main.css [hunk 13]
```diff
@@ -1 +1,228 @@
...
+
+.activeLink{
+    color: rgb(151, 220, 61);
+}
+.navbar-nav>li>a:focus,.navbar-nav>li>a:hover{
+    color: rgb(151, 210, 61);
+    background-color: rgb(21,100,156) ;
+}
+
+.procedureMessage {
```
Comment (raj-vlabs) [inline experiment/simulation/css/main.css]: use the hyphenated naming convention for css consistently. CamelCase should be used for js. So this class will be named as procedure-message

File: experiment/simulation/css/main.css [hunk 14]
```diff
@@ -1 +1,228 @@
...
+    z-index: -1;
+    position: absolute;
+    align-self: center; 
+    object-fit: cover;
+    top: 10vw;
+    width: 78vw;
+    left: 10vw;
+}
+
+.spoon1, .spoon2, .spoon3{
```
Comment (raj-vlabs) [inline experiment/simulation/css/main.css]: Why do we need three spoon classes here? Wouldn't one spoon class be sufficient?

File: experiment/simulation/css/main.css [hunk 15]
```diff
@@ -1 +1,228 @@
...
+.detergent1,.detergent2,.detergent3{
+    position: absolute;
+    align-self: center;
+    object-fit: cover;
+    top: 3vw;
+    right: 19vw;
+    width: 6vw;
+}
+
+.spoon2, .spoon3, .detergent2, .detergent3{
```
Comment (raj-vlabs) [inline experiment/simulation/css/main.css]: Create a separate class for opacity named as is-opaque or is-transparent or is-not-opaque. That way you'd be able to reuse that class as needed. This way you are just replacing the id based styling with classnames. It does not achieve the purpose of using class based styling.

File: experiment/simulation/js/main.js [hunk 16]
```diff
@@ -1 +1,258 @@
...
+}
+if(document.getElementsByClassName("detergent2")[0]){
+    document.getElementsByClassName("detergent2")[0].style.pointerEvents = 'none';
+}
+if(document.getElementsByClassName("spoon3")[0]){
+    document.getElementsByClassName("spoon3")[0].style.pointerEvents = 'none';
+}
+if(document.getElementsByClassName("detergent3")[0]){
+    document.getElementsByClassName("detergent3")[0].style.pointerEvents = 'none';
+}
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js:10]: Please create an array of all the element ids and add the pointer styling in a forEach loop

File: experiment/simulation/js/main.js [hunk 17]
```diff
@@ -1 +1,258 @@
...
+
+
+// Functions for structure of soap part.
+// below function shows information about micelle parts when hovered upon
+function showMicelleInformation(part){
+    'use strict';
+    let tailInfo = document.getElementsByClassName("micelle-information")[0];
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js]: this should be declared cosnt. cost is the default type, let is used only when using const gives error

File: experiment/simulation/js/main.js [hunk 18]
```diff
@@ -1 +1,258 @@
...
+// when cursor is not above them.
+function hideTailInformation(){
+    'use strict';
+    let tailInfo = document.getElementsByClassName("micelle-information")[0];
+    tailInfo.innerHTML = "";
+}
+
+function hideHeadInformation(){
+    'use strict';
+    let headInfo = document.getElementsByClassName("micelle-information")[0];
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js]: Why is headInfo coming from the same element as tailInfo?

File: experiment/simulation/js/main.js [hunk 19]
```diff
@@ -1 +1,258 @@
...
+    headInfo.innerHTML = "";
+}
+
+// Functions of Cleansing action
+// Below function moves the spoon and detergent from 
+// top right corner towards the beaker.
+function moveSpoon(){
+    'use strict';
+    let procedureMessage = document.getElementsByClassName("procedureMessage")[0]
+    let SCREENWIDTH = window.innerWidth
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js]: Why is the variable name in uppercase? It is not even declared a const.

File: experiment/simulation/js/main.js [hunk 20]
```diff
@@ -1 +1,258 @@
...
+    let changeYForDetergent = '24vw';
+    if(SCREENWIDTH > 1200){
+        changeYForDetergent = '19vw';
+    }
+
+    // disable onclick of spoon and detergent
+    document.getElementsByClassName("spoon"+detergentAddCounter)[0].style.pointerEvents = 'none';
+    document.getElementsByClassName("detergent"+detergentAddCounter)[0].style.pointerEvents = 'none';
```
Comment (raj-vlabs) [inline experiment/simulation/js/main.js]: If you are looking for a specific element, it is better to get it byId rather than getting an array byClassName and then taking the first element. Think about a scenario where someone enhances the experiment to add another element and wants to reuse the classes for styling

File: experiment/simulation/micelleFormation.html [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css" integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm" crossorigin="anonymous">
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
+    <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js" integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl" crossorigin="anonymous"></script>
+    <link rel="stylesheet" href="./css/main.css">
+    <title>Formation of micelle</title>
+
+</head>
+    <body>
```
Comment (raj-vlabs) [inline experiment/simulation/micelleFormation.html:15]: Check the indentation. Head and Body should be at the same level of indentation.

**Review by raj-vlabs — CHANGES_REQUESTED** [review]
1. Look at the feasibility of lazy loading the videos to have better perceived performance.
2. Use hyphenated names for CSS classes. CamelCase for js
3. Rethink the choice of styling classes
4. Use const by default and let only when const does not work
5. Differentiate cases where you should getEementbyId or getElementsByClassName

File: experiment/simulation/cleansingAction.html (hunk 10 above)
Comment (NitinChandak) [reply experiment/simulation/cleansingAction.html]: It wasn't necessary. I just copied it by mistake.

File: experiment/simulation/cleansingAction.html (hunk 11 above)
Comment (NitinChandak) [reply experiment/simulation/cleansingAction.html:46]: We could do that but then that video will not be in sync with the experiment. The size of the largest video file I used was 254 Kb. If you want, I can implement lazy loading the video.

File: experiment/simulation/css/main.css (hunk 15 above)
Comment (NitinChandak) [reply experiment/simulation/css/main.css]: Keeping opacity in the same classes will help in anime animation.

**Review by raj-vlabs — APPROVED** [review]
I'd suggest to implement lazy-loading and see how it impacts the experiment. However, I will not hold the review for that.
